docs/instrumentation/tracing-java-interop.md

```diff
@@ -0,0 +1,208 @@
+
+To interoperate with Java libraries that rely on the OpenTelemetry Java context, you need to activate the context manually.
+The following utility method allows you to extract the current otel4s context and set it into the ThreadLocal variable:
+
+```scala mdoc:silent:reset
+import cats.effect.Sync
+import cats.mtl.Local
+import cats.syntax.flatMap._
+import org.typelevel.otel4s.java.context.Context
+import io.opentelemetry.context.{Context => JContext}
+
+def useJContext[F[_]: Sync, A](use: JContext => A)(implicit L: Local[F, Context]): F[A] = 
+  Local[F, Context].ask.flatMap { ctx => // <1>
+    Sync[F].delay {
+      val jContext: JContext = ctx.underlying // <2>
+      val scope = jContext.makeCurrent() // <3>
+      try {
+        use(jContext)
+      } finally {
+        scope.close()
+      }
+    }
+  }
+```
+
+1) `Local[F, Context].ask` - get the current otel4s context  
+2) `ctx.underlying` - unwrap otel4s context and get `JContext`  
+3) `jContext.makeCurrent()` - activate `JContext` within the current thread
+
+**Note:** we use `Sync[F].delay` to handle the side effects.
+Depending on your use case, you may prefer `Sync[F].interruptible` or `Sync[F].blocking`.
+ 
+Now we can run a slightly modified original 'problematic' example:
+```scala
+tracer.span("test").use { span => // start 'test' span using otel4s
+  IO.println(s"Otel4s ctx: ${span.context}") >> useJContext[IO, Unit] { _ =>
+    val jSpanContext = JSpan.current().getSpanContext // get a span from the ThreadLocal variable
+     println(s"Java ctx: $jSpanContext") 
+  }
+}
+```
+
+The output will be:
+```
+Java ctx: {traceId=06f5d9112efbe711947ebbded1287a30, spanId=26ed80c398cc039f, ...}
+Otel4s ctx: {traceId=06f5d9112efbe711947ebbded1287a30, spanId=26ed80c398cc039f, ...}
+```
+
+As we can see, the tracing information is in sync now,
+and you can use Java-instrumented libraries within the `useJContext` block. 
+
+## Pekko HTTP example
+
+[PekkoHttpExample][pekko-http-example] is a complete example that shows how to use otel4s
+with OpenTelemetry Java instrumented libraries.
+
+[opentelemetry-java]: https://github.com/open-telemetry/opentelemetry-java
+[opentelemetry-java-autoconfigure]: https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md
+[opentelemetry-java-context]: https://github.com/open-telemetry/opentelemetry-java/blob/main/context/src/main/java/io/opentelemetry/context/Context.java
+[opentelemetry-java-span]: https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/trace/Span.java
+[otel4s-context]: https://github.com/typelevel/otel4s/blob/main/java/common/src/main/scala/org/typelevel/otel4s/java/context/Context.scala
+[cats-mtl-local]: https://typelevel.org/cats-mtl/mtl-classes/local.html
+[pekko-http]: https://pekko.apache.org/docs/pekko-http/current
+[pekko-http-example]: https://github.com/typelevel/otel4s/blob/main/examples/src/main/scala/PekkoHttpExample.scala
```

Comment:
should these link to tags or explicit commits (you can get the latter by pressing `y` on your keyboard on the page for any file in the repo) so that files can't move out from under us and break links?

Reply:
Well, showing outdated information is better than showing a broken link 😄